nixos/modules/services/misc/sonarr.nix

```diff
@@ -0,0 +1,45 @@
+{ config, pkgs, lib, mono, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sonarr;
+in
+{
+  options = {
+    services.sonarr = {
+      enable = mkEnableOption "Sonarr";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.sonarr = {
+      description = "Sonarr";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        test -d /var/lib/sonarr/ || {
+          echo "Creating sonarr data directory in /var/lib/sonarr/"
+          mkdir -p /var/lib/sonarr/
+        }
+        chown -R sonarr /var/lib/sonarr/
+        chmod 0700 /var/lib/sonarr/
+      '';
+
+      serviceConfig = {
+        Type = "simple";
+        User = "sonarr";
+        Group = "nogroup";
+        PermissionsStartOnly = "true";
+        ExecStart = "${pkgs.mono}/bin/mono ${pkgs.sonarr}/bin/NzbDrone.exe";
+        Restart = "on-failure";
+      };
+    };
+
+    users.extraUsers.sonarr = {
+      name = "sonarr";
```

Comment:
Note that if `name` is left unset, the attribute name is used instead, providing a slight reduction in boilerplate.